src/screens/MainApp/DetailedDiagnose/components/NewComment/index.js

```diff
@@ -0,0 +1,42 @@
+import React, { useState } from 'react'
+import { View, TextInput } from 'react-native'
+import { PrimaryButton, GrayButton, Description } from '~/components'
+import { theme } from '~/constants'
+import styles from './styles'
+
+const NewComment = ({ onNewComment }) => {
+  const [comment, setComment] = useState('')
+  const [inputEnabled, setInputEnabled] = useState(true)
+
+  const sendEnabled = (comment.length > 0)
+  const handlePress = async () => {
+    const savedComment = comment.slice()
+    setInputEnabled(false)
+    setComment('')
+    if (onNewComment) await onNewComment(savedComment)
+    setInputEnabled(true)
+  }
+
+  return (
+    <View style={styles.container}>
+      <TextInput
+        onChangeText={setComment}
+        value={comment}
+        style={styles.input}
+        editable={inputEnabled}
+        placeholder='Responder...'
+        placeholderTextColor={theme.colors.gray}
+      />
+      {sendEnabled &&
+        <PrimaryButton onPress={handlePress}>
+          <Description white>Enviar</Description>
+        </PrimaryButton>}
+      {!sendEnabled &&
+        <GrayButton>
+          <Description white>Enviar</Description>
+        </GrayButton>}
```

Comment:
I think this is a common pattern that is repeated a lot on the whole app. What do you think about adding a `disabled` boolean property to `PrimaryButton`? Internally, if `disabled` is `true`, it would render as a `GrayButton` and remove the `onPress` callback.

cc @MPucheta

Reply:
I think it is a good idea, I also think that there won't be disabled buttons with other color than gray.

Reply:
I'm gonna make a change to make this short

Reply:
It's a good idea. I thought about creating a "ToggleButton" but this solution is better.